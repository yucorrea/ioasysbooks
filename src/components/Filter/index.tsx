import React, { useState, useCallback } from 'react';
import { useDispatch } from 'react-redux';
import { useTheme } from 'styled-components/native';
import Icon from 'react-native-vector-icons/Ionicons';
import styled, { css } from 'styled-components/native';

import { Button } from './../Button';
import { ModalComponent } from '../Modal';

import { years, categories } from './../../../__mocks__';

import { GET_BOOKS } from '../../store/slices/booksSlice';

export function Filter() {
  const [search, setSearch] = useState('');
  const [selectedCategory, setSelectedCategory] = useState<string>('');
  const [selectedYear, setSelectedYear] = useState<number | null>(null);
  const [modal, setModal] = useState(false);

  const { colors } = useTheme();
  const dispatch = useDispatch();

  const handleInputSearch = useCallback(() => {
    dispatch(GET_BOOKS({ search, page: 1 }));
  }, [dispatch, search]);

  const handleApplyFilters = useCallback(() => {
    console.log({ selectedCategory, selectedYear })
    setModal(false);
  }, [dispatch, selectedCategory, selectedYear]);

  return (
    <>
      <StyledContainer>
        <StyledInputContainer>
          <StyledInput
            placeholder="Procure um livro"
            placeholderTextColor={colors.text}
            value={search}
            onChangeText={e => setSearch(e)}
            onSubmitEditing={handleInputSearch}
          />
          <StyledSearchButton onPress={handleInputSearch}>
            <Icon name="search-outline" size={24} color={colors.title} />
          </StyledSearchButton>
        </StyledInputContainer>

        <StyledOptionButton onPress={() => setModal(true)}>
          <Icon color={colors.title} name="options-outline" size={24} />
        </StyledOptionButton>
      </StyledContainer>

      <ModalComponent show={modal} close={() => setModal(false)}>
        <StyledTitle>Selecione a categoria</StyledTitle>
        <StyledOptionsFilter>
          {categories.map(option => (
            <StyledFilterButton
              key={option.id.toString()}
              selected={option.value === selectedCategory}
              onPress={() => setSelectedCategory(option.value)}
            >
              <StyledFilterButtonText selected={option.value === selectedCategory}>
                {option.value}
              </StyledFilterButtonText>
            </StyledFilterButton>
          ))}
        </StyledOptionsFilter>

        <StyledTitle>Selecione o ano</StyledTitle>
        <StyledOptionsFilter>
          {years.map(option => (
            <StyledFilterButton
              key={option.id.toString()}
              selected={option.value === selectedYear}
              onPress={() => setSelectedYear(option.value)}
            >
              <StyledFilterButtonText selected={option.value === selectedYear}>
                {option.value}
              </StyledFilterButtonText>
            </StyledFilterButton>
          ))}
        </StyledOptionsFilter>

        <StyledButtonFilter onPress={handleApplyFilters} title="Filtar" />
      </ModalComponent>
    </>
  );
}

interface FilterButtonProps {
  selected?: boolean;
}

const StyledContainer = styled.View`
  width: 100%;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  margin: 32px 0px;
`;

const StyledInputContainer = styled.View`
  flex: 1;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin-right: 22px;
  border-radius: 2px;
  height: 48px;
  border: 1px solid rgba(51, 51, 51, 0.2);
`;

const StyledInput = styled.TextInput`
  flex: 1;
  font-family: ${({ theme }) => theme.fonts.medium};
  color: rgba(153, 153, 153, 0.7);
  height: 48px;
  font-size: 12px;
  margin-left: 11px;
`;

const StyledSearchButton = styled.TouchableOpacity`
  margin-right: 20px;
`;

const StyledOptionButton = styled.TouchableOpacity``;

const StyledButtonFilter = styled(Button)`
  align-self: center;
  border: 1px solid ${({ theme }) => theme.colors.primary};
`;

const StyledOptionsFilter = styled.View`
  width: 100%;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  margin-top: 8px;
  margin-bottom: 36px;
`;

const StyledFilterButton = styled.TouchableOpacity<FilterButtonProps>`
  border: 1px solid
    ${({ selected, theme }) =>
      selected ? theme.colors.title : 'rgba(51, 51, 51, 0.3)'};

  border-radius: 44px;
  justify-content: center;
  align-items: center;

  ${({ selected, theme }) =>
    selected &&
    css`
      background: ${theme.colors.title};
    `}

  margin: 8px 8px 8px 0px;
`;

const StyledFilterButtonText = styled.Text<FilterButtonProps>`
  font-family: ${({ theme }) => theme.fonts.regular};
  font-size: 12px;
  padding: 6px 16px;

  color: ${({ selected, theme }) =>
    selected ? theme.colors.background : theme.colors.title};
`;

const StyledTitle = styled.Text`
  font-size: 12px;
  font-family: ${({ theme }) => theme.fonts.medium};
  color: ${({ theme }) => theme.colors.title};
`;
