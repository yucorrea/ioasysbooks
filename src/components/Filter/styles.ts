import styled, { css } from 'styled-components/native';

interface FilterButtonProps {
  selected?: boolean;
}

import { Button } from './../Button';

export const Container = styled.View`
  width: 100%;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  margin: 32px 0px;
`;

export const InputContainer = styled.View`
  flex: 1;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin-right: 22px;
  border-radius: 2px;
  height: 48px;
  border: 1px solid rgba(51, 51, 51, 0.2);
`;

export const Input = styled.TextInput`
  flex: 1;
  font-family: ${({ theme }) => theme.fonts.medium};
  color: rgba(153, 153, 153, 0.7);
  height: 48px;
  font-size: 12px;
  margin-left: 11px;
`;

export const SearchButton = styled.TouchableOpacity`
  margin-right: 20px;
`;

export const OptionButton = styled.TouchableOpacity``;

export const ButtonFilter = styled(Button)`
  align-self: center;
  border: 1px solid ${({ theme }) => theme.colors.primary};
`;

export const OptionsFilter = styled.View`
  width: 100%;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  margin-top: 8px;
  margin-bottom: 36px;
`;

export const FilterButton = styled.TouchableOpacity<FilterButtonProps>`
  border: 1px solid
    ${({ selected, theme }) =>
      selected ? theme.colors.title : 'rgba(51, 51, 51, 0.3)'};

  border-radius: 44px;
  justify-content: center;
  align-items: center;

  ${({ selected, theme }) =>
    selected &&
    css`
      background: ${theme.colors.title};
    `}

  margin: 8px 8px 8px 0px;
`;

export const FilterButtonText = styled.Text<FilterButtonProps>`
  font-family: ${({ theme }) => theme.fonts.regular};
  font-size: 12px;
  padding: 6px 16px;

  color: ${({ selected, theme }) =>
    selected ? theme.colors.background : theme.colors.title};
`;

export const Title = styled.Text`
  font-size: 12px;
  font-family: ${({ theme }) => theme.fonts.medium};
  color: ${({ theme }) => theme.colors.title};
`;

export const Icon = styled.Image``;
