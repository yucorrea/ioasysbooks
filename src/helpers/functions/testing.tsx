import React from 'react';
import renderer from 'react-test-renderer';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';

import theme from '../../config/theme';

export const matchSnapshotWithTheme = (component :  React.ReactElement) => {
  const tree = renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>).toJSON();
  expect(tree).toMatchSnapshot();
};

export const renderWithTheme = (component : React.ReactElement, options?: RenderOptions) => {
  return render(<ThemeProvider theme={theme}> {component} </ThemeProvider>, options);
};
