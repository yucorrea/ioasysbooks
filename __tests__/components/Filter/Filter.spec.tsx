import React from 'react';
import { fireEvent } from '@testing-library/react-native';
import Filter from '@components/Filter';
import { matchSnapshotWithTheme, renderWithTheme } from '@helpers/functions/testing';

describe('Filter component', () => {
  it('matches snapshots', () => {
    const snap = matchSnapshotWithTheme(<Filter />);
    expect(snap).toMatchSnapshot();
  });

  it('render correctly', () => {
    renderWithTheme(<Filter />);
  });

  it('value change input filter', () => {
    const { getByTestId } = renderWithTheme(<Filter />);

    const inputSearch = getByTestId('input-search');
    fireEvent.changeText(inputSearch, 'Ab');
    expect(inputSearch.props.value).toEqual('Ab');
  });

  it('filter when click search button', () => {
    const { getByTestId } = renderWithTheme(<Filter />);

    const buttonSearch = getByTestId('button-search');
    const inputSearch = getByTestId('input-search');

    fireEvent.changeText(inputSearch, 'Book');
    fireEvent.press(buttonSearch);

    expect(inputSearch.props.value).toHaveLength(4);
  })


  it('check if has open modal when click icon', () => {
    const { getByTestId } = renderWithTheme(<Filter />);

    const buttonShowModal = getByTestId('button-show-modal');
    const filterModal = getByTestId('modal-component');

    fireEvent.press(buttonShowModal);
    expect(filterModal.props.visible).toBeTruthy();
  })
})
