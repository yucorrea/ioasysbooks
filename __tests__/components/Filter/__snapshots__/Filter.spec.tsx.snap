// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component matches snapshots 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 32,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 32,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "rgba(51, 51, 51, 0.2)",
            "borderRadius": 2,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "height": 48,
            "justifyContent": "space-between",
            "marginRight": 22,
          },
        ]
      }
    >
      <TextInput
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Procure um livro"
        placeholderTextColor="#999999"
        style={
          Array [
            Object {
              "color": "#999999",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontFamily": "Heebo-Medium",
              "fontSize": 12,
              "height": 48,
              "marginLeft": 11,
            },
          ]
        }
        testID="input-search"
        value=""
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginRight": 20,
            "opacity": 1,
          }
        }
        testID="button-search"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#333333",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="button-show-modal"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#333333",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    testID="modal-component"
    transparent={true}
    visible={false}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 0, 0, 0.5)",
            "height": 1334,
            "justifyContent": "center",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": 750,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 4,
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
              "width": "100%",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-end",
                "width": "100%",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(51, 51, 51, 0.2)",
                "borderRadius": 16,
                "borderStyle": "solid",
                "borderWidth": 1,
                "height": 32,
                "justifyContent": "center",
                "opacity": 1,
                "width": 32,
              }
            }
            testID="button-modal-close"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontSize": 16,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "Heebo-Medium",
                "fontSize": 12,
              },
            ]
          }
        >
          Selecione a categoria
        </Text>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "marginBottom": 36,
                "marginTop": 8,
                "width": "100%",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(51, 51, 51, 0.2)",
                "borderRadius": 44,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 8,
                "marginTop": 8,
                "opacity": 1,
              }
            }
          >
            <Text
              selected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 12,
                    "paddingBottom": 6,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 6,
                  },
                ]
              }
            >
              Design
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(51, 51, 51, 0.2)",
                "borderRadius": 44,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 8,
                "marginTop": 8,
                "opacity": 1,
              }
            }
          >
            <Text
              selected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 12,
                    "paddingBottom": 6,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 6,
                  },
                ]
              }
            >
              UX Design
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(51, 51, 51, 0.2)",
                "borderRadius": 44,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 8,
                "marginTop": 8,
                "opacity": 1,
              }
            }
          >
            <Text
              selected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 12,
                    "paddingBottom": 6,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 6,
                  },
                ]
              }
            >
              UI Design
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(51, 51, 51, 0.2)",
                "borderRadius": 44,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 8,
                "marginTop": 8,
                "opacity": 1,
              }
            }
          >
            <Text
              selected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 12,
                    "paddingBottom": 6,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 6,
                  },
                ]
              }
            >
              Arquitetura da Informação
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(51, 51, 51, 0.2)",
                "borderRadius": 44,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 8,
                "marginTop": 8,
                "opacity": 1,
              }
            }
          >
            <Text
              selected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 12,
                    "paddingBottom": 6,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 6,
                  },
                ]
              }
            >
              CSS
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(51, 51, 51, 0.2)",
                "borderRadius": 44,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 8,
                "marginTop": 8,
                "opacity": 1,
              }
            }
          >
            <Text
              selected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 12,
                    "paddingBottom": 6,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 6,
                  },
                ]
              }
            >
              Usabilidade
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(51, 51, 51, 0.2)",
                "borderRadius": 44,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 8,
                "marginTop": 8,
                "opacity": 1,
              }
            }
          >
            <Text
              selected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 12,
                    "paddingBottom": 6,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 6,
                  },
                ]
              }
            >
              Design Thinking
            </Text>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "Heebo-Medium",
                "fontSize": 12,
              },
            ]
          }
        >
          Selecione o ano
        </Text>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "marginBottom": 36,
                "marginTop": 8,
                "width": "100%",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(51, 51, 51, 0.2)",
                "borderRadius": 44,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 8,
                "marginTop": 8,
                "opacity": 1,
              }
            }
          >
            <Text
              selected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 12,
                    "paddingBottom": 6,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 6,
                  },
                ]
              }
            >
              2015
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(51, 51, 51, 0.2)",
                "borderRadius": 44,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 8,
                "marginTop": 8,
                "opacity": 1,
              }
            }
          >
            <Text
              selected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 12,
                    "paddingBottom": 6,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 6,
                  },
                ]
              }
            >
              2016
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(51, 51, 51, 0.2)",
                "borderRadius": 44,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 8,
                "marginTop": 8,
                "opacity": 1,
              }
            }
          >
            <Text
              selected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 12,
                    "paddingBottom": 6,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 6,
                  },
                ]
              }
            >
              2017
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(51, 51, 51, 0.2)",
                "borderRadius": 44,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 8,
                "marginTop": 8,
                "opacity": 1,
              }
            }
          >
            <Text
              selected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 12,
                    "paddingBottom": 6,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 6,
                  },
                ]
              }
            >
              2018
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(51, 51, 51, 0.2)",
                "borderRadius": 44,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 8,
                "marginTop": 8,
                "opacity": 1,
              }
            }
          >
            <Text
              selected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 12,
                    "paddingBottom": 6,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 6,
                  },
                ]
              }
            >
              2019
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(51, 51, 51, 0.2)",
                "borderRadius": 44,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 8,
                "marginTop": 8,
                "opacity": 1,
              }
            }
          >
            <Text
              selected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 12,
                    "paddingBottom": 6,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 6,
                  },
                ]
              }
            >
              2020
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(51, 51, 51, 0.2)",
                "borderRadius": 44,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 8,
                "marginTop": 8,
                "opacity": 1,
              }
            }
          >
            <Text
              selected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 12,
                    "paddingBottom": 6,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 6,
                  },
                ]
              }
            >
              2021
            </Text>
          </View>
        </View>
        <View
          animatedStyle={
            Object {
              "value": Object {
                "height": 36,
                "width": 86,
              },
            }
          }
          collapsable={false}
          style={
            Object {
              "height": 36,
              "width": 86,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#B22E6F",
                "borderRadius": 44,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "opacity": 1,
              }
            }
          >
            <Text
              animatedStyle={
                Object {
                  "value": Object {
                    "opacity": 1,
                  },
                }
              }
              collapsable={false}
              style={
                Object {
                  "color": "#B22E6F",
                  "fontFamily": "Heebo-Medium",
                  "fontSize": 16,
                  "opacity": 1,
                }
              }
            >
              Filtar
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>,
]
`;

exports[`Filter component matches snapshots 2`] = `undefined`;
